internal/tests/api/hostsets/host_set_test.go

```diff
@@ -246,27 +253,24 @@ func TestCrud(t *testing.T) {
 	require.NoError(err)
 
 	h, err = hClient.Create(tc.Context(), c.Item.Id, hostsets.WithName("foo"),
-		hostsets.WithAttributes(map[string]interface{}{"foo": "bar"}), hostsets.WithPreferredEndpoints([]string{"dns:test"}))
+		hostsets.WithAttributes(map[string]interface{}{"foo": "bar"}), hostsets.WithPreferredEndpoints([]string{"dns:test"}),
+		hostsets.WithSyncIntervalSeconds(-1))
 	require.NoError(err)
 	assert.Equal("foo", h.Item.Name)
 	assert.Equal(uint32(1), h.Item.Version)
 
 	h, err = hClient.Read(tc.Context(), h.Item.Id)
 	require.NoError(err)
 	assert.Equal("foo", h.Item.Name)
-	assert.Equal(uint32(1), h.Item.Version)
+	// If the plugin syncs after creation, its version will increment, so it can either be 1 or 2
+	assert.Contains([]uint32{1, 2}, h.Item.Version)
 
-	h, retried := retryableUpdate(hClient, h.Item.Id, h.Item.Version, hostsets.WithName("bar"),
+	h, _ = retryableUpdate(hClient, h.Item.Id, h.Item.Version, hostsets.WithName("bar"),
```

Comment:
Since we are not using the bool returned from retryable update and doing a contains check instead, can you update the function definition to be something like:
```
retryableUpdate := func(c *hostsets.Client, hcId string, version uint32, opts ...hostsets.Option) *hostsets.HostSetUpdateResult {
```